### ISSUE TITLE
`lld-link: error: duplicate symbol` (between libcxxabi and libcxx) on Windows for C++ with `-O2`

### ISSUE THREAD

**Issue by polkovnikov** [ziglang/zig#10126]
### Remember to search before filing a new report

- [X] I searched for this bug and did not find it in the issue tracker, and furthermore, the title I used above will make this new bug report turn up in the search results for my query.

### Zig Version

0.9.0-dev.1588+9bb7ff68c

### Steps to Reproduce

If on Windows 10 (64-bit) I write simple C++ hello world program:

```
#include <iostream>
int main() { std::cout << "Hello, World!" << std::endl; }
```

and compile it with

`c:/bin/zig/zig.exe c++ -std=c++11 -O2 -m64 hello.cpp`

### Expected Behavior

Compile should be successful, without errors.

### Actual Behavior

When compiling with `-O2` then I get lots of following duplicate-symbol errors.

Same error happens for `-O1`, `-O2`, `-O3`, but no such error (successful compile) for `-O0`.

```
lld-link: error: duplicate symbol: operator delete(void*, std::align_val_t)
>>> defined at C:\bin\zig\lib\libcxxabi\src/stdlib_new_delete.cpp
>>>            c++abi.lib(stdlib_new_delete.obj)
>>> defined at C:\bin\zig\lib\libcxx\src/new.cpp
>>>            c++.lib(new.obj)

lld-link: error: duplicate symbol: operator delete(void*, std::align_val_t, std::nothrow_t const&)
>>> defined at C:\bin\zig\lib\libcxxabi\src/stdlib_new_delete.cpp
>>>            c++abi.lib(stdlib_new_delete.obj)
>>> defined at C:\bin\zig\lib\libcxx\src/new.cpp
>>>            c++.lib(new.obj)

lld-link: error: duplicate symbol: operator delete(void*, unsigned long long, std::align_val_t)
>>> defined at C:\bin\zig\lib\libcxxabi\src/stdlib_new_delete.cpp
>>>            c++abi.lib(stdlib_new_delete.obj)
>>> defined at C:\bin\zig\lib\libcxx\src/new.cpp
>>>            c++.lib(new.obj)

lld-link: error: duplicate symbol: operator delete[](void*, std::align_val_t)
>>> defined at C:\bin\zig\lib\libcxxabi\src/stdlib_new_delete.cpp
>>>            c++abi.lib(stdlib_new_delete.obj)
>>> defined at C:\bin\zig\lib\libcxx\src/new.cpp
>>>            c++.lib(new.obj)

lld-link: error: duplicate symbol: operator delete[](void*, std::align_val_t, std::nothrow_t const&)
>>> defined at C:\bin\zig\lib\libcxxabi\src/stdlib_new_delete.cpp
>>>            c++abi.lib(stdlib_new_delete.obj)
>>> defined at C:\bin\zig\lib\libcxx\src/new.cpp
>>>            c++.lib(new.obj)

lld-link: error: duplicate symbol: operator delete[](void*, unsigned long long, std::align_val_t)
>>> defined at C:\bin\zig\lib\libcxxabi\src/stdlib_new_delete.cpp
>>>            c++abi.lib(stdlib_new_delete.obj)
>>> defined at C:\bin\zig\lib\libcxx\src/new.cpp
>>>            c++.lib(new.obj)

...................... And others .................................
```

**Comment by nuald** [comment]
Related to LTO: #8531 

For now, please use `-fno-lto` when compiling for Windows:

```sh
zig.exe c++ -fno-lto -std=c++11 -O2 -m64 hello.cpp
```

**Comment by polkovnikov** [comment]
@nuald Thanks! `-fno-lto` solved the issue. Can we, Windows users, expect in the future that default variant (with LTO) will also be fixed later?

**Comment by andrewrk** [comment]
Fixed with eecb669074f7798975366ac834ae25fa185f05fa. I also made it so that LTO is disabled by default for Windows when linking libc++ until a few other issues can be resolved.
